File: Modeling .NET/VB/Bank/Bank/ClientGenerator.vb
Imports Simulation

Public Class ClientGenerator
    Inherits Process

    Protected Overrides Sub Execute()
        Dim i As Integer
        For i = 0 To Params.MaxClientCount - 1
            ClearFinished()
            Dim clt As New Client
            clt.Activate()
            Hold(BankSimulation.RandClient.Exponential(Params.MeanClientInterval))
        Next
    End Sub
End Class

File: Modeling .NET/CSharp/3.1.1 Bank/Bank/ClientGenerator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Simulation;

namespace Bank
{
    /// <summary>
    /// Генератор клиентов банка
    /// </summary>
    class ClientGenerator : Process
    {
        /// <summary>
        /// Алгоритм работы генератора
        /// </summary>
        protected override void Execute()
        {
            // Количество клиентов задано заранее
            for (int i = 0; i < Param.MaxClientCount; i++)
            {
                ClearFinished();
                // Создать клиента и запустить его
                (new Client()).Activate();
                // ОЖидать перед созданием следующего
                Hold(BankSimulation.RandClient.Exponential(Param.MeanClientInterval));
            }
        }
    }
}
